### PR TITLE
Put all errors into the context instead of throwing them, so they won…

### PR DESCRIPTION
**Pull request by bserdar**
…'t be lost

### CODE REVIEW

File: crud/src/main/java/com/redhat/lightblue/crud/mongo/MongoCRUDController.java [hunk 1]
```diff
@@ -349,11 +347,10 @@ public CRUDFindResponse find(CRUDOperationContext ctx,
                 ctx.addError(Error.get(MongoCrudConstants.ERR_NO_ACCESS, "find:" + ctx.getEntityName()));
             }
         } catch (Error e) {
-            // rethrow lightblue error
-            throw e;
+            ctx.addError(e);
```
Comment (paterczm) [inline crud/src/main/java/com/redhat/lightblue/crud/mongo/MongoCRUDController.java]: Why isn't the error logged (LOGGER.error) here? It's very helpful to have all errors logged lightblue side, clients may not log them.

Comment (bserdar) [inline crud/src/main/java/com/redhat/lightblue/crud/mongo/MongoCRUDController.java]: We have logging in some of the Error.get methods. Lets add logging to all
Error.get methods, so these would be logged everywhere.

Comment (paterczm) [inline crud/src/main/java/com/redhat/lightblue/crud/mongo/MongoCRUDController.java]: Are Error.get always called when an error happens? If so and they are all supposed to log, then statements
   LOGGER.error(e.getMessage(), e);
in catch are redundant.

Comment (bserdar) [inline crud/src/main/java/com/redhat/lightblue/crud/mongo/MongoCRUDController.java]: Yes, all Error's are propagated back to client. Anything that is internal
to the system is throw using regular exceptions, and if they propagate,
they are eventually put into an Error.

I'll remove the redundant logs, and add logging to Error.get

**Pull request merged**